Text/TextQualityWatchdog/Util/TikaExtract.py

```diff
@@ -0,0 +1,159 @@
+
+        # Get the blob client for the current blob
+        blob_client = cont_client.get_blob_client(blobname)
+
+        # Get the path to save the unstructured file to
+        download_file_path = APP_ROOT_STR + "\\" + DOWNLOAD_DIR + "\\" + filename
+
+        # Get the full path to the destination file
+        extracted_file_path = APP_ROOT_STR + "\\" + DEST_DIR + "\\" + filename
+
+        # Get the full path to the metadata file
+        metadata_file_path = APP_ROOT_STR + "\\" + METADATA_DIR + "\\" + filename
+
+        # Download the blob to a local file
+        with open(download_file_path, "wb") as download_file:
+            download_file.write(blob_client.download_blob().readall())
+        print("Downloaded blob to \n\t" + download_file_path)
+
+        # Invoke tika server to perform extraction
+        parsedFile = parser.from_file(download_file_path)
+
+        # Skip file if fail to extract text
+        if (parsedFile['content'] is None) or (type(parsedFile['content']) != str):
+            os.remove(download_file_path)
+            print("No text extracted, file skipped.\n")
+
+            # Update high water mark
+            with open(last_read_file_path, "w") as last_read_file:
+                last_read_file.write(filename)
+
+            # Update metrics and save to disk
+            metrics_dict['total'] += 1
+            metrics_dict['failed'] += 1
+            with open(metrics_dict_file_path, 'wb') as picklefile:
+                pickle.dump(metrics_dict, picklefile)
+
+            continue
+
+        # Write the extracted text to the destination
+        with io.open(extracted_file_path, "w", encoding="utf-8") as f:
+            f.write(parsedFile['content'])
+
+        # Invoke tika to detect language of the text file
+        lang = language.from_file(extracted_file_path)
+
+        # Remove non-English text files
+        if lang != "en":
+            os.remove(extracted_file_path)
+            os.remove(download_file_path)
+            print("Non-English text file removed.\n")
+
+            # Update metrics and save to disk
+            metrics_dict['total'] += 1
+            metrics_dict['nonEng'] += 1
+            with open(metrics_dict_file_path, 'wb') as picklefile:
+                pickle.dump(metrics_dict, picklefile)
+
+        else:
+            # Serialize metadata and dump in pickle file
+            metadata_dict = parsedFile['metadata']
+            with open(metadata_file_path + '.pkl', 'wb') as picklefile:
+                pickle.dump(metadata_dict, picklefile)
+
+            # Update metrics and save to disk
+            metrics_dict['total'] += 1
+            metrics_dict['success'] += 1
+            with open(metrics_dict_file_path, 'wb') as picklefile:
+                pickle.dump(metrics_dict, picklefile)
+
+            print("Success!\n")
+
+        # Update high water mark
+        with open(last_read_file_path, "w") as last_read_file:
+            last_read_file.write(filename)
```

Comment:
Would it be useful to put a write lock on that file while the process is running?

Reply:
Thanks for the suggestion Bertrand! I wrote this helper script with single-threading in mind, and I believe that the "with open" statement is conventional in Python since it creates a context manager to handle file I/O correctly.